client/my-sites/pages/main.jsx

```diff
@@ -134,6 +158,7 @@ class PagesMain extends React.Component {
 }
 
 const mapState = state => ( {
+	state,
```

Comment:
We should expose only what's needed from the state rather than the whole thing. Otherwise the component will re-render every time anything in the state changes.

In this case, I think we only need the search placeholder:

```suggestion
	searchPlaceholder: getPostTypeLabel( ... ),
```

Reply:
I know, I faced this dilemma when I was implementing.
Basically the problem is I am using `query.type` and `state` as a parameter of `getPostTypeLabel`, and query is not defined in the props. 

I will move the query definition in the props then.